spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -59,12 +59,13 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
     logWarning(s"Comet native execution is disabled due to: $reason")
   }
 
-  def supportedDataType(dt: DataType): Boolean = dt match {
+  def supportedDataType(dt: DataType, allowComplex: Boolean = false): Boolean = dt match {
```

Comment:
just thinking do we really need `allowComplex`, which more looks like a `supportComplexTypes` or we can use the Comet param like its done for operator/expressions.

Moreover the isSchemaSupported doesn't have this flag which causes some inconsistency imho

Reply:
Yes there's a lot of oddities in how supported data types works right now. It should really just be operator/expression dependent that gets bubbled up to whether the whole plan is supported or not. 

I'm not sure what you mean by 
>or we can use the Comet param like its done for operator/expressions.

Reply:
Like 
```
  private[comet] def isCometAllOperatorEnabled(conf: SQLConf): Boolean = {
    COMET_EXEC_ALL_OPERATOR_ENABLED.get(conf)
  }
```

Reply:
It'd be a little odd to have a config for struct type support IMO, since either the code supports it or it doesn't. Maybe you would want to override to disable struct support, but you would still need all the same code to support it as well